src/main/java/goodgenerator/blocks/tileEntity/FuelRefineFactory.java

```diff
@@ -217,8 +217,13 @@ protected CheckRecipeResult validateRecipe(@NotNull GT_Recipe recipe) {
             @Override
             protected GT_OverclockCalculator createOverclockCalculator(@NotNull GT_Recipe recipe,
                     @NotNull GT_ParallelHelper helper) {
-                return super.createOverclockCalculator(recipe, helper)
-                        .setSpeedBoost(1F / (float) (1 << (Tier - recipe.mSpecialValue)));
+                long consumedPower = (long) recipe.mEUt * (1L << (Tier - recipe.mSpecialValue));
+                return new GT_OverclockCalculator().setRecipeEUt(consumedPower)
+                        .setDuration(
+                                (int) Math.ceil(
+                                        (recipe.mDuration * helper.getDurationMultiplierDouble())
```

Comment:
I don't think duration multiplier should be applied here?

Reply:
True idk what I thought here.